index.ts

```diff
@@ -0,0 +1,72 @@
+import { spawn, SpawnOptionsWithoutStdio } from "child_process";
+import type { Plugin as VitePlugin } from "vite";
+
+// Utility to invoke a given sbt task and fetch its output
+function printSbtTask(task: string, cwd?: string): Promise<string> {
+  const args = ["--batch", "-no-colors", "-Dsbt.supershell=false", `print ${task}`];
+  const options: SpawnOptionsWithoutStdio = {
+    cwd: cwd,
+  };
+  const child = process.platform === 'win32'
+    ? spawn("sbt.bat", args.map(x => `"${x}"`), {shell: true, ...options})
+    : spawn("sbt", args, options);
+
+  let fullOutput: string = '';
+
+  child.stdout.setEncoding('utf-8');
+  child.stdout.on('data', data => {
+    fullOutput += data;
+    process.stdout.write(data); // tee on my own stdout
+  });
+
+  child.stderr.setEncoding('utf-8');
+  child.stderr.on('data', data => {
+    process.stderr.write(data); // tee on my own stderr
+  });
+
+  return new Promise((resolve, reject) => {
+    child.on('close', code => {
```

Comment:
Does this event fire if the process cannot be spawned at all?

Reply:
No! Good catch. I added an `'error'` handler for that.